engine/src/io/data_parser/OrcParser.cpp

```diff
@@ -68,5 +69,21 @@ void orc_parser::parse_schema(
 	}
 }
 
+std::unique_ptr<ral::frame::BlazingTable> orc_parser::get_metadata(std::vector<data_handle> handles,
+	std::vector<std::shared_ptr<arrow::io::RandomAccessFile>> files, int offset) {
+	std::vector<size_t> num_stripes(files.size());
+	std::vector<cudf::io::parsed_orc_statistics> statistics(handles.size());
+	for(size_t file_index = 0; file_index < handles.size(); file_index++) {
+		std::string file_path = handles[file_index].uri.toString(true);
+		statistics[file_index] = cudf::io::read_parsed_orc_statistics(cudf::io::source_info{file_path});
```

Comment:
Here we dont want to use file_path, because if its a file on s3 or hdfs, it wont work. You need to do the RandorAccessFile `files`

Reply:
done in the _3e6af7d_  commit